pkg/cloud/libvirt/client/domain.go

```diff
@@ -109,7 +109,10 @@ func newDevicesDef(virConn *libvirt.Connect) *libvirtxml.DomainDeviceList {
 	// Both "s390" and "s390x" are linux kernel architectures for Linux on IBM z Systems, and they are for 31-bit and 64-bit respectively.
 	// Graphics/Spice isn't supported on s390/s390x platform.
 	// Same case for PowerPC systems as well
-	if !strings.HasPrefix(arch, "s390") && !strings.HasPrefix(arch, "ppc64") {
+	switch arch {
+	case "s390", "s390x", "ppc64", "ppc64le", "aarch64":
```

Comment:
On which architectures *is* SPICE supported?  I'm wondering if this shouldn't be changed so that new architectures don't need to continually be added here.

Reply:
i thought it's better to blacklist so we don't miss any arches. https://www.spice-space.org/faq.html claims support for arm, but libvirt still threw an error saying SPICE graphics is not supported for qemu, so not sure about that.

But yes - this is not ideal - maybe it would be better to just enable it for x86_64. @cfergeau thoughts?

Reply:
You have upstream support, and Red Hat support, which can be different things. On RHEL, I see that it's built for x86_64 and aarch64. I think there has been work to make spice support optional, so it may depend on which packages are installed.
Since we also add a console device just before the spice graphics element, it's probably fine to only add it on x86_64 for now. If the console is not enough for debugging purposes, in the future, we can add a vnc graphics element on platforms where spice is disabled.

Reply:
yup..so looks like qemu-kvm in RHEL is not compiled with spice for non-x86..just looked at the spec file in the rpm:
```
%ifarch x86_64
    %global kvm_target    x86_64
%else
    %global have_spice   0
    %global have_opengl  0
    %global have_gluster 0
```

i can put a request to have it enabled for aarch64, but in the meanwhile i think I will change the conditional here to only support it only for x86.

Reply:
i will open a separate PR for this